newsfragments/2670.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+:func:`inspect.iscoroutinefunction` and the like now give correct answers when
+called on KI-protected functions.
```

Comment:
Could we add a test for this

Reply:
Ah I assumed this was already tested by the whole `@_public` machinery

I can work on a dedicated test for this

Reply:
pushed